TM1py/Services/ProcessService.py

```diff
@@ -194,9 +194,15 @@ def execute_process_with_return(self, process: Process, **kwargs) -> Tuple[bool,
         """
         Run unbound TI code directly
         :param process: a TI Process Object
+        :param kwargs: dictionary of process parameters and values
         :return: success (boolean), status (String), error_log_file (String)
         """
         url = "/api/v1/ExecuteProcessWithReturn?$expand=*"
+        if kwargs:
+            for parameter_name, parameter_value in kwargs.items():
+                process.add_parameter(name=parameter_name,
```

Comment:
I think you will need to remove the parameter first before you can add it.

Reply:
Something like?

```python
        if kwargs:
            for parameter_name, parameter_value in kwargs.items():
                if parameter_name in process.parameters:
                    process.remove_parameter(name=parameter_name)
                process.add_parameter(name=parameter_name,
                                      prompt=parameter_name,
                                      value=parameter_value)
```

Reply:
Actually the remove_parameter does not result in error if the parameter does not exist. Not sure if this is better or not

Therefore

```python
                process.remove_parameter(name=parameter_name)
                process.add_parameter(name=parameter_name,
                                      prompt=parameter_name,
                                      value=parameter_value)
```